src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -436,6 +470,27 @@ def decode(self, codes: torch.Tensor) -> torch.Tensor:
             quantized_out = quantized_out + quantized
         return quantized_out
 
+    def forward(self, embeddings: torch.Tensor, bandwidth: Optional[float] = None):
+        residual = embeddings
+        num_quantizers = self.get_num_quantizers_for_bandwidth(bandwidth)
+
+        quantized_out = torch.tensor(0.0, device=embeddings.device)
+        all_losses = []
+        all_indices = []
+
+        for layer in self.layers[:num_quantizers]:
+            loss, embed_ind, quantized = layer(residual)
+            # Note: The source from FB research did not detach the quantized here. There may have been a bug in the
```

Comment:
Interesting! My only fear with diverging from the original implementations is that we can't follow the hyper parameters given in the original implementation, e.g.
1. If a user has a working training set-up in the original codebase, they won't necessarily have a working one in Transformers
2. If we want to give sensible starting hyper parameters, we can't use the ones given in the paper -> we need to do our own HP search
3. If our implementation doesn't work, it's going to be hard asking the model authors how to fix this!

Maybe we can open an issue and check whether this is indeed a bug in the original?

Reply:
Good point. I tend to agree that it would be better to follow the original implementation, even if there is a bug with it. The original issue in the FB research source is tracked here: https://github.com/facebookresearch/encodec/issues/25. I was able to reproduce the residual commitment losses being 0 in the latest code.

Reply:
FYI looks like this hasn't been fixed even in the updated audiocraft code:
https://github.com/facebookresearch/audiocraft/blob/a2b96756956846e194c9255d0cdadc2b47c93f1b/audiocraft/quantization/core_vq.py#L374C14-L374C14